Sources/TuistGenerator/Generator/ConfigGenerator.swift

```diff
@@ -284,35 +285,36 @@ final class ConfigGenerator: ConfigGenerating {
     }
 
     private func deploymentTargetDerivedSettings(target: Target) -> SettingsDictionary {
-        guard let deploymentTarget = target.deploymentTarget else {
-            return [:]
-        }
-
         var settings: SettingsDictionary = [:]
-
-        switch deploymentTarget {
-        case let .iOS(version, devices):
-            var deviceFamilyValues: [Int] = []
-            if devices.contains(.iphone) { deviceFamilyValues.append(1) }
-            if devices.contains(.ipad) { deviceFamilyValues.append(2) }
-
-            settings["TARGETED_DEVICE_FAMILY"] = .string(deviceFamilyValues.map { "\($0)" }.joined(separator: ","))
-            settings["IPHONEOS_DEPLOYMENT_TARGET"] = .string(version)
-
-            if devices.contains(.ipad), devices.contains(.mac) {
-                settings["SUPPORTS_MACCATALYST"] = "YES"
-                settings["DERIVE_MACCATALYST_PRODUCT_BUNDLE_IDENTIFIER"] = "YES"
-            } else {
-                // Unless explicitly specified, when the platform the Product is a framework, these default to YES.
-                settings["SUPPORTS_MACCATALYST"] = "NO"
+        var deviceFamilyValues: [Int] = []
+
+        for deploymentTarget in target.deploymentTargets {
+            switch deploymentTarget {
+            case let .iOS(version, devices):
+                if devices.contains(.iphone) { deviceFamilyValues.append(1) }
+                if devices.contains(.ipad) { deviceFamilyValues.append(2) }
+                
+                settings["IPHONEOS_DEPLOYMENT_TARGET"] = .string(version)
+                
+                if devices.contains(.ipad), devices.contains(.mac) {
+                    settings["SUPPORTS_MACCATALYST"] = "YES"
+                    settings["DERIVE_MACCATALYST_PRODUCT_BUNDLE_IDENTIFIER"] = "YES"
+                } else {
+                    // Unless explicitly specified, when the platform the Product is a framework, these default to YES.
+                    settings["SUPPORTS_MACCATALYST"] = "NO"
+                }
+            case let .macOS(version):
+                settings["MACOSX_DEPLOYMENT_TARGET"] = .string(version)
+            case let .watchOS(version):
+                deviceFamilyValues.append(4)
+                settings["WATCHOS_DEPLOYMENT_TARGET"] = .string(version)
+            case let .tvOS(version):
+                deviceFamilyValues.append(3)
+                settings["TVOS_DEPLOYMENT_TARGET"] = .string(version)
             }
-        case let .macOS(version):
-            settings["MACOSX_DEPLOYMENT_TARGET"] = .string(version)
-        case let .watchOS(version):
-            settings["WATCHOS_DEPLOYMENT_TARGET"] = .string(version)
-        case let .tvOS(version):
-            settings["TVOS_DEPLOYMENT_TARGET"] = .string(version)
         }
+        
+        settings["TARGETED_DEVICE_FAMILY"] = .string(deviceFamilyValues.map { "\($0)" }.joined(separator: ","))
```

Comment:
So this is what makes the device multi-platform?

Reply:
No, It doesn't. it defines devices/simulators that we can select as destination in Xcode. **SUPPORTED_PLATFORMS**    allows multiple platforms for target . for example SUPPORTED_PLATFORMS = "iphoneos iphonesimulator macosx";